trino/sqlalchemy/dialect.py

```diff
@@ -336,15 +336,20 @@ def has_sequence(self, connection: Connection, sequence_name: str, schema: str =
         """Trino has no support for sequence. Returns False indicate that given sequence does not exists."""
         return False
 
-    def _get_server_version_info(self, connection: Connection) -> Any:
-        query = "SELECT version()"
-        try:
-            res = connection.execute(sql.text(query))
-            version = res.scalar()
-            return tuple([version])
-        except exc.ProgrammingError as e:
-            logger.debug(f"Failed to get server version: {e.orig.message}")
-            return None
+    @classmethod
+    def _get_server_version_info(cls, connection: Connection) -> Any:
+        def get_server_version_info(_):
+            query = "SELECT version()"
+            try:
+                res = connection.execute(sql.text(query))
+                version = res.scalar()
+                return tuple([version])
+            except exc.ProgrammingError as e:
+                logger.debug(f"Failed to get server version: {e.orig.message}")
+                return None
+
+        # We make the server_version_info be evaluated lazily
+        cls.server_version_info = property(get_server_version_info, lambda instance, value: None)
```

Comment:
Why not use the decorator `@property`?

Reply:
Because `server_version_info` is not in our codebase but in sqlalchemy's, so we need to resort to this type of instantiation.

Reply:
I mean why use the `property()` function instead of a decorator.

Reply:
Have a look at [sqlalchemy's code](https://github.com/sqlalchemy/sqlalchemy/blob/eb286c15f096771dbb128acbe8fe03e94aa72f6a/lib/sqlalchemy/engine/default.py#L514). Simply adding the property wouldn't work as it would be overwritten.

Reply:
Thanks, I get it know. But it seems we're doing a hack here where we override the server info property with one that has a getter for lazy evaluation, but a noop setter. Then we return a None, which sqlalchemy tries to assign to that prop, but that'll get ignored because go the setter. Is it worth adding a comment explaining this, if I got it right?